src/Fieldset/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import newId from '../utils/newId';
+import styles from './Fieldset.scss';
+import ValidationMessage from '../ValidationMessage/index';
+import Variant from '../utils/constants';
+
+const inputProps = {
+  variantIconDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  children: PropTypes.node,
+  className: PropTypes.arrayOf(PropTypes.string),
+  id: PropTypes.string,
+  isValid: PropTypes.bool,
+  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  invalidMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  variant: PropTypes.shape({
```

Comment:
Are there plans to add more `variant` options in the future? A `shape` with one field feels a little over-structured...

Reply:
I didn't create Variant, so I'm not 100% sure what the purpose of it is. It looks like it was designed so that there would be different Variant "types". The only one right now is `status`: `INFO`, `WARNING`, `DANGER`, `SUCCESS`. But, I don't know what other Variant types are possible and don't want to collapse this to a string if we'd want to add more variants later.

Reply:
What do you mean by

> don't want to collapse this to a string if we'd want to add more variants later

Couldn't the props just be `buttonVariant`, `inputVariant`, `anotherInputVariant`, etc. (assuming we want multiple variants).